hifi_qt.py

```diff
@@ -27,40 +27,100 @@ class QtDownloader:
     def __init__(self, args):
         self.args = args
         self.configFilePath = os.path.join(args.build_root, 'qt.cmake')
-        self.version = os.getenv('VIRCADIA_USE_QT_VERSION', '5.12.3')
-
         self.assets_url = hifi_utils.readEnviromentVariableFromFile(args.build_root, 'EXTERNAL_BUILD_ASSETS')
 
-        defaultBasePath = os.path.expanduser('~/hifi/qt')
-        self.basePath = os.getenv('HIFI_QT_BASE', defaultBasePath)
-        if (not os.path.isdir(self.basePath)):
-            os.makedirs(self.basePath)
-        self.path = os.path.join(self.basePath, self.version)
-        self.fullPath = os.path.join(self.path, 'qt5-install')
-        self.cmakePath = os.path.join(self.fullPath, 'lib/cmake')
+        # OS dependent information
+        system = platform.system()
+
+        qt_found = False
+
+        # Here we handle the 3 possible cases of dealing with Qt:
+        if os.getenv('VIRCADIA_USE_SYSTEM_QT'):
+            # 1. Using the system provided Qt. This is only recommended for Qt 5.15.0 and above,
+            # as it includes a required fix on Linux.
+            #
+            # This path only works on Linux as neither Windows nor OSX ship Qt.
+
+            if system != "Linux":
+                raise Exception("Using the system Qt is only supported on Linux")
+
+            cmake_paths = [ "lib64/cmake", "lib/cmake" ]
```

Comment:
Hardcoding the architecture doesn't seem like a great solution, but I had to add this for Debian Sid:
```patch
-            cmake_paths = [ "lib64/cmake", "lib/cmake" ]
+            cmake_paths = [ "lib64/cmake", "lib/cmake", "lib/x86_64-linux-gnu/cmake" ]
```
Normally cmake knows to add the system cmake path automatically, if it is possible to not  pass it at all that might be preferable.